hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAdmin2.java

```diff
@@ -864,4 +865,44 @@ public void testSlowLogResponses() throws Exception {
     Assert.assertEquals(onlineLogRecords.size(), 0);
   }
 
+  @Test
+  public void testGetRegionServers() throws Exception {
+    // get all live server names
+    List<ServerName> serverNames = new ArrayList<>(ADMIN.getRegionServers(true));
+    Assert.assertEquals(3, serverNames.size());
+
+    List<ServerName> serversToDecom = new ArrayList<>();
+    ServerName serverToDecommission = serverNames.get(0);
+
+    serversToDecom.add(serverToDecommission);
+    ADMIN.decommissionRegionServers(serversToDecom, false);
+    waitForServerCommissioned(serverToDecommission, true);
+
+    Assert.assertEquals(2, ADMIN.getRegionServers(true).size());
+    Assert.assertEquals(3, ADMIN.getRegionServers(false).size());
+
+    ADMIN.recommissionRegionServer(serverToDecommission, Collections.emptyList());
+    waitForServerCommissioned(null, false);
+
+    Assert.assertEquals(3, ADMIN.getRegionServers(true).size());
+    Assert.assertEquals(3, ADMIN.getRegionServers(false).size());
+  }
+
+  private static void waitForServerCommissioned(ServerName excludeServer,
+      boolean anyServerDecommissioned) {
+    TEST_UTIL.waitFor(3000, () -> {
```

Comment:
Is the timeout chosen randomly?

Reply:
Not randomly, but it is(3s) more than enough for de/re-commission to take place. Should be fine?

Reply:
I have used the same before also for decommission test.